pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -430,7 +439,11 @@ cdef class {{name}}HashTable(HashTable):
             for i in range(n):
                 val = values[i]
 
-                if ignore_na and (
+                if ignore_na and use_mask:
```

Comment:
if we're worried about perf for existing cases, could take this check outside of the loop?

Reply:
We need to check the mask for each value inside the loop, so not sure what can be moved outside?

Reply:
I was referring to the use_mask check; it would basically become a separate loop or even method

Reply:
I don't think duplicating the full loop is worth it (the loop itself is 40 lines below here), given the minor performance impact I showed in the timings.